### PR TITLE
Reinit binding and optionChanged functions in a nested component after changes in component

### PR DESCRIPTION
**Pull request by dxvladislavvolkov**

### CODE REVIEW

File: src/core/configuration-component.ts [hunk 1]
```diff
@@ -19,15 +19,23 @@ interface IConfigurable extends IConfigurationOwner {
     $_config: Configuration;
 }
 
+function $_initBinding(vueInstance) {
```
Comment (tar-z) [inline src/core/configuration-component.ts]: I would suggest to remove `$_` prefix from the function name. It is required only for custom private properties

File: src/core/component.ts [hunk 2]
```diff
@@ -73,6 +73,9 @@ const BaseComponent: VueConstructor<IBaseComponent> = Vue.extend({
     },
 
     updated() {
+        this.$children.forEach((child) => {
```
Comment (IlyaKhD) [inline src/core/component.ts]:
```suggestion
        this.$children.forEach(initBinding);
```

File: src/core/configuration-component.ts [hunk 3]
```diff
@@ -19,15 +19,23 @@ interface IConfigurable extends IConfigurationOwner {
     $_config: Configuration;
 }
 
+function initBinding(vueInstance) {
+    if (vueInstance.$vnode) {
```
Comment (IlyaKhD) [inline src/core/configuration-component.ts]:
```suggestion
    if (!vueInstance.$vnode) return;
```

File: src/core/configuration-component.ts [hunk 4]
```diff
@@ -19,15 +19,23 @@ interface IConfigurable extends IConfigurationOwner {
...
+        const componentOptions = (vueInstance.$vnode.componentOptions as any as IConfigurable);
+        const config = componentOptions && componentOptions.$_config;
+        if (config) {
```
Comment (IlyaKhD) [inline src/core/configuration-component.ts]:
```suggestion
        if (!config) return;
```

**Pull request merged**